### PR TITLE
Add kernel driver for MPU9250 Gyroscope/Accelerometer/Magnetometer Sensor.

### PR DESCRIPTION
**Pull request by wurmmi**
Add a kernel driver module that interacts with an FPGA core, which provides data from the MPU9250 Gyroscope/Accelerometer/Magnetometer sensor.

### CODE REVIEW

File: drivers/mpu9250/mpu9250.c [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  tmpBuf.mag_x = 0x7777;
+  tmpBuf.mag_y = 0x8888;
+  tmpBuf.mag_z = 0x9999;
+  tmpBuf.timestamp_lo = 0x47001100;
+  tmpBuf.timestamp_hi = 0x48001200;
+  /* ------ TESTING ONLY --------------------------------------------------------- */
+
+  /* copy data from kernel space buffer into user space */
+  if (count > 0)
+    count = count - copy_to_user(buf, (char *)&dev->buffer + *offp, count);
```
Comment (wurmmi) [inline drivers/mpu9250/mpu9250.c]: not sure whether the part with `(char*)&dev->buffer` is correct.. :thinking:

Comment (skaupper) [reply drivers/mpu9250/mpu9250.c]: as long as `dev->buffer` is the struct, and not a pointer to it, you should be fine

**Comment by skaupper** [comment]
it could be a problem that we are not able to test it until we have a device with the requested name in the device tree :D

**Comment by wurmmi** [comment]
exactly.. we are kind of forced to wait on @pagdot

**Pull request merged**